packages/edit-post/src/components/layout/index.js

```diff
@@ -209,21 +209,19 @@ function Layout( {
 				showIconLabels: get( 'core', 'showIconLabels' ),
 				isDistractionFree: get( 'core', 'distractionFree' ),
 				showMetaBoxes:
+					! DESIGN_POST_TYPES.includes( currentPostType ) &&
```

Comment:
This and the change at L351 are the effectual changes, the rest are merely due to refactoring for reuse of the return value from `useNavigateToEntityRecord` instead of selecting it again from the editor store.